api/router.go

```diff
@@ -4,17 +4,55 @@ import (
 	"github.com/gin-gonic/gin"
 	ctx "github.com/smira/aptly/context"
 	"net/http"
+	"strconv"
+	"os/exec"
+	"encoding/json"
 )
 
 var context *ctx.AptlyContext
 
+// middleware to track API calls and call a hook script
+func ApiHooks(cmd string, conf string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Next()
+
+		params , _ := json.Marshal(c.Params)
+		query, _   := json.Marshal(c.Request.URL.Query())
+
+		env := []string{
+			("APTLY_METHOD="      + c.Request.Method),
+			("APTLY_REQ_URL="     + c.Request.URL.String()),
+			("APTLY_REMOTE_ADDR=" + c.Request.RemoteAddr),
+			("APTLY_STATUS="      + strconv.Itoa(c.Writer.Status())),
+			("APTLY_CONFIG="      + conf),
+			("APTLY_PARAMS="      + string(params)),
+			("APTLY_QUERY="       + string(query)),
+		}
+
+		cmd := exec.Command(cmd)
+		cmd.Env = env
+
+		// fire and forget
+		go func(){
+			cmd.Run()
```

Comment:
I think logging errors is required here, otherwise failures to run hook script would go completely unnoticed?